src/game_display.cpp

```diff
@@ -556,17 +556,25 @@ void game_display::highlight_reach(const pathfind::paths &paths_list)
 	highlight_another_reach(paths_list);
 }
 
-void game_display::highlight_another_reach(const pathfind::paths &paths_list)
+void game_display::highlight_another_reach(const pathfind::paths &paths_list,
+			const map_location& goal)
 {
 	// Fold endpoints of routes into reachability map.
 	for (const pathfind::paths::step &dest : paths_list.destinations) {
 		reach_map_[dest.curr]++;
 	}
 	reach_map_changed_ = true;
+
+	if(goal != map_location::null_location() && paths_list.destinations.contains(goal))
+	{
+		const map_location enemy_unit_location = paths_list.destinations.get_path(paths_list.destinations.find(goal))[0];
+		units_that_can_reach_goal_.emplace(enemy_unit_location);
```

Comment:
Use `insert()` instead.

Reply:
Changed. What's the difference in this case? Not calling the `map_location` constructor if `enemy_unit_location` is already an element of the set?

Reply:
There is no semantical difference. It's just that `emplace()` is intended for constructing objects in-place instead of copying them.